### PR TITLE
More layering and compile strictness

### PR DESCRIPTION
**Pull request by anthonydresser**
Adds another layering requirement for platform and fixes all the hygiene issues in platform.

- Moves a bunch of interfaces in the modelview stuff from workbench to platform
- Duplicates some of the connection types used in the extension api into platform
- Moves some of the workbench specific theme stylers to workbench
- Replaces rxjs with events in some classes
- Use custom types for generic types coming from angular
- Moves some services from platform to workbench. They weren't being used in platform anywhere else but were using code from workbench.

### CODE REVIEW

File: src/sql/platform/connection/common/interfaces.ts [hunk 1]
```diff
@@ -18,3 +18,22 @@ export interface IConnectionProfileStore {
 	id: string;
 }
 
+export enum ServiceOptionType {
```
Comment (anthonydresser) [inline src/sql/platform/connection/common/interfaces.ts]: Copied these from the sqlexthost types. Could probably remove those types eventually.

File: src/sql/platform/dashboard/browser/interfaces.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IDisposable } from 'vs/base/common/lifecycle';
+
+export interface IComponentEventArgs {
```
Comment (anthonydresser) [inline src/sql/platform/dashboard/browser/interfaces.ts]: Moves a bunch of these interfaces into platform from workbench

File: src/sql/platform/theme/common/styler.ts [hunk 3]
```diff
@@ -270,22 +255,3 @@ export function attachCheckboxStyler(widget: IThemable, themeService: IThemeServ
 		disabledCheckboxForeground: (style && style.disabledCheckboxForeground) || sqlcolors.disabledCheckboxForeground
 	}, widget);
 }
-
-export function attachPanelStyler(widget: IThemable, themeService: IThemeService) {
```
Comment (anthonydresser) [inline src/sql/platform/theme/common/styler.ts]: Moved these stylers to workbench.

File: src/sql/platform/angularEventing/browser/angularEventingService.ts [hunk 4]
```diff
@@ -35,7 +35,7 @@ export interface IAngularEventingService {
 	 * @param uri Uri of the dashboard
 	 * @param cb Listening function
```
Comment (Charles-Gagnon) [inline src/sql/platform/angularEventing/browser/angularEventingService.ts]: Remove doc param

**Pull request merged**